django/db/backends/mysql/base.py

```diff
@@ -360,6 +360,8 @@ def check_constraints(self, table_names=None):
                     )
 
     def is_usable(self):
+        if self.connection is None:
+            return False
```

Comment:
Yes, that's what I had in mind. This is why I suggested the idea of including this logic in the base implementation (to eliminate having to copy the logic into three different places).

Alternatively, you could do something like this at the beginning:

    try:
        conn = self.connection
    except AttributeError:
        return False